cpp/memilio/math/adapt_rk.h

```diff
@@ -208,19 +208,22 @@ class RKIntegratorCore : public IntegratorCore<FP>
      * @param[in,out] t current time
      * @param[in,out] dt current time step size h=dt
      * @param[out] ytp1 approximated value y(t+1)
+     * @param[in] force_step_size Forces a step with exactly size dt.
      */
     bool step(const DerivFunction<FP>& f, Eigen::Ref<Eigen::VectorXd const> yt, double& t, double& dt,
-              Eigen::Ref<Eigen::VectorXd> ytp1) const override
+              Eigen::Ref<Eigen::VectorXd> ytp1, const bool force_step_size = false) const override
     {
         assert(0 <= m_dt_min);
         assert(m_dt_min <= m_dt_max);
```

Comment:
IMO this is a perfect use case for for an assert, since we *cannot* continue if these conditions fail, but checking this every iteration is needlessly expensive.
And since the asserts are printed in plain text on failure, I don't see the need to add text to it.